diff --git a/plugin/defaults.vim b/plugin/defaults.vim
index c9400ad..8429a64 100644
--- ./plugin/defaults.vim
+++ ./plugin/defaults.vim
@@ -163,9 +163,9 @@ if !exists('g:formatdef_astyle_java')
     if filereadable('.astylerc')
         let g:formatdef_astyle_java = '"astyle --mode=java --options=.astylerc"'
     elseif filereadable(expand('~/.astylerc')) || exists('$ARTISTIC_STYLE_OPTIONS')
-        let g:formatdef_astyle_java = '"astyle --mode=java"'
+        let g:formatdef_astyle_java = '"astyle --mode=java --options=' . expand('~/loadrc/' . s:hostname . '.astylerc') . '"'
     else
-        let g:formatdef_astyle_java = '"astyle --mode=java --style=java -pcH".(&expandtab ? "s".shiftwidth() : "t")'
+        let g:formatdef_astyle_java = '"astyle --mode=java -nfxejoUys4pHU --style=allman -pcH".(&expandtab ? "s".&shiftwidth : "t")'
     endif
 endif
 
