diff --git a/plugin/defaults.vim b/plugin/defaults.vim
index 26c67f9..c934d76 100644
--- ./plugin/defaults.vim
+++ ./plugin/defaults.vim
@@ -196,9 +196,9 @@ if !exists('g:formatdef_astyle_java')
     if filereadable(astylerc)
         let g:formatdef_astyle_java = '"astyle --mode=java --options=' . astylerc . '"'
     elseif exists('$ARTISTIC_STYLE_OPTIONS')
-        let g:formatdef_astyle_java = '"astyle --mode=java"'
+        let g:formatdef_astyle_java = '"astyle --mode=java --options=' . expand('~/loadrc/' . s:hostname . '.astylerc') . '"'
     else
-        let g:formatdef_astyle_java = '"astyle --mode=java --style=java -pcH".(&expandtab ? "s".shiftwidth() : "t")'
+        let g:formatdef_astyle_java = '"astyle --mode=java -nfxejoUys4pHU --style=allman -pcH".(&expandtab ? "s".&shiftwidth : "t")'
     endif
 endif
 
